--- shishi-1.0.0/src/starttls.c~	2010-04-29 08:43:49.000000000 +0200
+++ shishi-1.0.0/src/starttls.c	2011-09-02 20:27:19.587913703 +0200
@@ -36,6 +36,7 @@
   gnutls_x509_crt cert;
   size_t i;
   int rc;
+  unsigned int status;
 
   cert_list = gnutls_certificate_get_peers (session, &cert_list_size);
   if (!cert_list)
@@ -195,8 +196,8 @@
 
   gnutls_x509_crt_deinit (cert);
 
-  rc = gnutls_certificate_verify_peers (session);
-  if (rc != GNUTLS_E_SUCCESS)
+  rc = gnutls_certificate_verify_peers2 (session, &status);
+  if (rc < 0)
     syslog (LOG_ERR, "TLS client certificate verify failed (%d): %s",
 	    rc, gnutls_strerror (rc));
 }
