--- shishi-1.0.1/configure.ac~	2012-03-12 19:18:21.000000000 +0100
+++ shishi-1.0.1/configure.ac	2012-11-21 13:57:49.229400717 +0100
@@ -24,7 +24,7 @@
 AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADERS(config.h)
-AM_INIT_AUTOMAKE([1.10 -Wall -Werror -Wno-override])
+AM_INIT_AUTOMAKE([1.10 -Wall -Wno-override])
 AM_SILENT_RULES([yes])
 
 # Library code modified:                              REVISION++
